python/lsst/sims/movingObjects/orbits.py

```diff
@@ -203,76 +206,88 @@ def assignSed(self, orbits, randomSeed=None):
                     skiprows += 1
-                    valuesheader = values
-            skiprows -= 1
+                    headervalues = values
             file.close()
 
-            if skiprows == -1:
-                # No header; assume it's a typical DES file - but is format KEP or COM?
-                names_COM = ('objId', 'FORMAT', 'q', 'e', 'i', 'node', 'argperi', 't_p',
-                             'H',  'epoch', 'INDEX', 'N_PAR', 'MOID', 'COMPCODE')
-                names_KEP = ('objId', 'FORMAT', 'a', 'e', 'i', 'node', 'argperi', 'meanAnomaly',
-                             'H', 'epoch', 'INDEX', 'N_PAR', 'MOID', 'COMPCODE')
-                orbits = pd.read_table(orbitfile, delim_whitespace=True, skiprows=0,
-                                       names=names_COM)
-                if orbits['FORMAT'][0] == 'KEP':
-                    orbits.columns = names_KEP
-
-            else:
+            if headervalues is not None:  # (and skiprows > -1)
                 # There is a header, but we also need to check if there is a comment key at the start
                 # of the proper header line.
-                linestart = valuesheader[0]
+                # ... Because this varies as well, and is sometimes separated from header columns.
+                linestart = headervalues[0]
                 if linestart == '#' or linestart == '!!' or linestart == '##':
-                    names = valuesheader[1:]
-                    skiprows += 1
-                # Read the data from disk.
-                if delim is None:
-                    orbits = pd.read_table(orbitfile, delim_whitespace=True, names=names, skiprows=skiprows)
+                    names = headervalues[1:]
                 else:
-                    orbits = pd.read_table(orbitfile, sep=delim, names=names, skiprows=skiprows)
+                    names = headervalues
+                # Add 1 to skiprows, so that we skip the header column line.
+                skiprows += 1
+
+        # So now skiprows is a value. If it is -1, then there is no header information.
+        if skiprows == -1:
+            # No header; assume it's a typical DES file -
+            # we'll assign the column names based on the FORMAT.
+            names_COM = ('objId', 'FORMAT', 'q', 'e', 'i', 'node', 'argperi', 't_p',
+                         'H',  'epoch', 'INDEX', 'N_PAR', 'MOID', 'COMPCODE')
+            names_KEP = ('objId', 'FORMAT', 'a', 'e', 'i', 'node', 'argperi', 'meanAnomaly',
+                         'H', 'epoch', 'INDEX', 'N_PAR', 'MOID', 'COMPCODE')
+            names_CART = ('objId', 'FORMAT', 'x', 'y', 'z', 'xdot', 'ydot', 'zdot',
+                          'H', 'epoch', 'INDEX', 'N_PAR', 'MOID', 'COMPCODE')
+            # First use names_COM, and then change if required.
+            orbits = pd.read_csv(orbitfile, delim_whitespace=True, header=None, names=names_COM)
+            if orbits['FORMAT'][0] == 'KEP':
```

Comment:
There isn't really a reason to not support mixed format except for not keeping this column for every object and then not checking the value when doing the conversion to/from oorb element format. For simplicity's sake, it's not -- but this is mostly because it's just not a use-case that's come up (usually people have elements that are all one or the other, for one type of object). 

My inclination would be to do the check and then throw an error .. if you think it'd important to keep the capability to handle mixed entries, then there are some more changes to make.

Reply:
Agreed that it's not important! But let's throw an exception (otherwise the code will silently do the wrong thing).